### PR TITLE
Base schnorr

### PR DESCRIPTION
**Pull request by gandrewstone**
This merges basic schnorr capabilities and tests into the lower level objects but does not implement or enable it on the blockchain.  Thanks to ABC devs and Mark Lundeberg for this work!

### CODE REVIEW

File: src/pubkey.cpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ static int ecdsa_signature_parse_der_lax(const secp256k1_context *ctx,
     }
 
     /* Integer tag byte for R */
-    if (pos == inputlen || input[pos] != 0x02)
+    if (pos >= inputlen || input[pos] != 0x02)
     {
```
Comment (ptschip) [inline src/pubkey.cpp]: why the change to "pos >=" rather than "pos ==" ?

**Review by ptschip — APPROVED** [review]
Did a spot check comparison with ABC...it looks all good.

**Comment by sickpig** [comment]
hey @markblundeberg would you mind have a look at this PR?

**Comment by markblundeberg** [comment]
Looks good! This is indeed the first step where Schnorr callables are made available in the main client code, but nothing consensus-related has been affected yet.

**Comment by sickpig** [comment]
Thanks @markblundeberg, I'm working on activation code right now. That code will be used to activate schnorr and segwit recovery features. 

will ping you again when the above will be ready.
